src/platform_specific/affinity.win.cc

```diff
@@ -0,0 +1,22 @@
+#include <cassert>
+
+#include <Windows.h>
+
+#include "affinity.h"
+#include "utils.h"
+
+namespace celerity {
+namespace detail {
+
+	uint32_t affinity_cores_available() {
+		using native_cpu_set = DWORD_PTR;
+
+		native_cpu_set affinity_base_mask;
+		[[maybe_unused]] native_cpu_set sys_affinity_mask;
+		auto base_mask_error = GetProcessAffinityMask(GetCurrentProcess(), &affinity_base_mask, &sys_affinity_mask);
+		assert(base_mask_error == 0 && "Error retrieving base affinity mask.");
+		return celerity::detail::util::popcount(affinity_base_mask);
```

Comment:
Since we're already in namespace `celerity::detail` here, this can be simplified to `util::popcount(affinity_base_mask);`